hack/deployer/runner/aks.go

```diff
@@ -175,24 +172,29 @@ func (d *AksDriver) configureDocker() error {
 		return err
 	}
 
-	cmd := `az aks show --resource-group {{.ResourceGroup}} --name {{.ClusterName}} --query "servicePrincipalProfile.clientId" --output tsv`
-	clientIds, err := NewCommand(cmd).AsTemplate(d.ctx).StdoutOnly().OutputList()
-	if err != nil {
-		return err
-	}
+	if !d.plan.ServiceAccount {
+		// it's already set for the ServiceAccount
```

Comment:
I wasn't sure what that comment meant. Maybe reversing the logic would help (and would also reduce the indentation level):

```go
if d.plan.ServiceAccount {
    // already configured
    return
}

cmd := ...
(...)
```

Reply:
Got it, will do separately.